woocommerce/woocommerce-android
# Fix orders badge inconsistency

Closes: #6080


### Description
Currently, the orders' badge is based on the number of orders in the app's database, this causes a major inconsistency due to two things:
1. It doesn't work well with pagination, as it would count only the fetched orders.
2. When an order is deleted, we delete just its summary from the DB, so it stays counted.

This PR attempts to fix these issues by using the API's endpoint `/wc/v3/reports/orders/totals` as the source of truth.

### Testing instructions
It would be great to have a site with a big number of orders to confirm the fix for the pagination issue, you can use this [plugin](https://wordpress.org/plugins/generate-random-orders-for-woocommerce/) for generating random orders.

And before starting the tests, please clear the app's data.

##### Case 1:
1. Open the app.
2. Confirm that the app shows the correct number of unfulfilled orders (orders with the status processing), and compare the number to what Woo Core shows in wp-admin.

##### Case 2: order status change from the app
1. Open the app.
2. Open an order, and toggle its status from `processing` (either enable it or disable it)
3. Confirm the badge is updated accordingly.

##### Case 3: order list fetch
1. Leave the app open
2. Open wp-admin, then toggle an order's status from `processing`.
3. Go back to the app, then refresh the order's list.
4. Confirm the badge is updated accordingly.

##### Case 4: refresh on push notification
1. Leave the app open.
2. Open the store then create an order.
3. Confirm that the badge is updated When the app receives a push notification.
Note: the order might not trigger a change, since orders with payment COD for example are created with status on_hold, in this case, check logcat and confirm the existence of the message: `New order notification received, re-check unfilled orders count`

### Images/gif
